dask/base.py

```diff
@@ -1345,34 +1361,67 @@ def register_pyarrow():
     def normalize_datatype(dt):
         return pickle.dumps(dt, protocol=4)
 
+    @normalize_token.register(pa.Table)
+    def normalize_table(dt):
+        return (
+            "pa.Table",
+            normalize_token(dt.schema),
+            normalize_token(dt.columns),
+        )
+
+    @normalize_token.register(pa.ChunkedArray)
+    def normalize_chunked_array(arr):
+        return (
+            "pa.ChunkedArray",
+            normalize_token(arr.type),
+            normalize_token(arr.chunks),
+        )
+
+    @normalize_token.register(pa.Array)
+    def normalize_chunked_array(arr):
+        return (
+            "pa.Array",
+            normalize_token(arr.type),
+            normalize_token(arr.buffers()),
+        )
+
+    @normalize_token.register(pa.Buffer)
+    def normalize_chunked_array(buf):
+        return ("pa.Buffer", hash_buffer_hex(buf))
```

Comment:
For perf reasons I ended up implementing a couple more special cases to avoid going through the "pickle multiple times" step for common objects like these.